front-end/src/components/filters/games-filter.tsx

```diff
@@ -33,43 +35,20 @@ export default function GamesFilter(props: Props) {
   return (
     <RisenBox1 title="Filters" sx={props.sx}>
       {
-        props.useSeason ? <FormControl sx={{minWidth: '300px', pt: 1, pb: 1}}>
-          <InputLabel id="season-filter-select-label">Season</InputLabel>
-          <Select
-            labelId="season-filter-select-label"
-            id="season-simple-select"
-            value={props.seasonConfig?.seasonId}
-            label="Season"
-            onChange={(event: SelectChangeEvent) => {props.seasonConfig?.setSeasonId(event.target.value)}}
-          >
-            {props.hideAllGames === true ? null : <MenuItem value="ALL">All Tournament Games</MenuItem>}
-            <MenuItem value="RISEN">All Risen Games</MenuItem>
-            {
-              props.seasonConfig?.seasons?.map(season => (
-                <MenuItem value={season.id}>{season.seasonName}</MenuItem>
-              ))
-            }
-          </Select>
-        </FormControl> : null
+        props.useSeason && <RisenSeasonSelector
+            callBack={(event: SelectChangeEvent) => {props.seasonConfig?.setSeasonId(event.target.value)}}
+            sx={{minWidth: '300px', pt: 1, pb: 1}}
+            seasonConfig={props.seasonConfig}
+            hideAllGames={props.hideAllGames !== undefined}/>
       }
       {
-        props.useRole ? <FormControl sx={{minWidth: '300px', pt: 1, pb: 1}}>
-        <InputLabel id="role-filter-select-label">Role</InputLabel>
-        <Select
-          labelId="role-filter-select-label"
-          id="role-simple-select"
-          value={props.roleConfig?.roleId}
-          label="Role"
-          onChange={(event: SelectChangeEvent) => {console.log(event.target.value); props.roleConfig?.setRoleId(GameRoles[event.target.value as keyof typeof GameRoles])}}
-        >
-          <MenuItem value={GameRoles.ALL}>All</MenuItem>
-          <MenuItem value={GameRoles.TOP}>Top</MenuItem>
-          <MenuItem value={GameRoles.JUNGLE}>Jungle</MenuItem>
-          <MenuItem value={GameRoles.MIDDLE}>Middle</MenuItem>
-          <MenuItem value={GameRoles.BOTTOM}>Bottom</MenuItem>
-          <MenuItem value={GameRoles.SUPPORT}>Support</MenuItem>
-        </Select>
-      </FormControl> : null
+          props.useRole && <RoleSelector
+              initalValue={props.roleConfig?.roleId}
+              sx={{minWidth: '300px', pt: 1, pb: 1}}
+              callBack={(event: SelectChangeEvent) => {
+                              console.log(event.target.value);
+                              props.roleConfig?.setRoleId(GameRoles[event.target.value as keyof typeof GameRoles])
```

Comment:
Yeah this is cursed typing.

Reply:
And yes, I do remember writing it.

Reply:
This is ur code KEKW

Reply:
Ik. Thats why I added the second comment. So you knew I wasn't flaming you.